File: patterns/Python/pattern9.py
"""
Generate the following pattern
1
0 1
1 0 1
0 1 0 1
1 0 1 0 1
"""

def pattern9(n):
    for i in range(1, n+1):
        if i%2 != 0:
            for j in range(i):
                if j%2 == 0:
                    print("1", end=" ")
                else:
                    print("0", end=" ")
                    
        else:
            for j in range(i):
                if j%2 != 0:
                    print("1", end=" ")
                else:
                    print("0", end=" ")
                    
        print("")
    
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern9(n)
    
if __name__ == "__main__":
    main()
           

File: patterns/C++/pattern9.cpp
/*
Generate the following pattern
1
0 1
1 0 1
0 1 0 1
1 0 1 0 1
*/

#include<bits/stdc++.h>
using namespace std;

void pattern9(int n){
    int start  = 1;
    for(int i=1; i<=n; i++){
        if(i%2 == 0) start = 0;
        else start = 1;

        for(int j=0; j<i; j++){
            cout << start;
            start = 1 - start;
        }
        cout << endl;
    }

}

int main(){
    int n;
    cout << "Enter the Number of rows: ";
    cin >> n;

    cout << "\nHere is the generated pattern\n";
    pattern9(n);
}